### ISSUE TITLE
[Solved] Virtualbox error in Manjaro: "Effective UID is not root"

### ISSUE THREAD

**Issue by Lemongrass38** [netblue30/firejail#1134]
Hi there! :)
First of all, thank you really much, netblue30, for developing Firejail! :) I really like it, and use it a lot. The issue: VirtualBox is not working in Manjaro 17.0 Xfce inside Firejail (nor firejail VirtualBox %U, nor firejail VirtualBox), I get the following error message:

> Effective UID is not root (euid=1000 egid=1000 uid=1000 gid=1000) (rc=-10)
> Please try reinstalling VirtualBox.
> where: SUPR3HardenedMain what: 2 VERR_PERMISSION_DENIED (-10) - Permission denied. 

It works outside Firejail.
More info:
I have the following packages related to VirtualBox:
```
$ pacman -Q | grep virtualbox
linux310-virtualbox-host-modules 5.1.14-2
linux49-virtualbox-host-modules 5.1.14-7
virtualbox 5.1.14-3
```
And this kernel:
```
$ uname -a
Linux lemongrass-pc 4.9.13-1-MANJARO #1 SMP PREEMPT Sun Feb 26 13:27:28 UTC 2017 x86_64 GNU/Linux
```

**Comment by Fred-Barclay** [comment]
@Lemongrass38 Great seeing you around here! :smile: 
What happens with `firejail --noprofile virtualbox`?

**Comment by netblue30** [comment]
Works fine on Debian stable with VirtualBox 5.0.24. Check if vbox kernel modules are loaded into the kernel when you start the computer:
`````
$ lsmod | grep vbox
vboxpci                24576  0 
vboxnetadp             28672  0 
vboxnetflt             28672  0 
vboxdrv               450560  3 vboxnetadp,vboxnetflt,vboxpci
`````
It is possible the version you have tries to load the kernel modules and is prevented by firejail.

You can also try after a computer startup to start VirtualBox without a sandbox, vbox manager comes up and loads the kernel modules, than you shut it down and start it in the sandbox.

**Comment by Lemongrass38** [comment]
Hi, thank you both for the reply! ;)

Unfortunately after the day I posted this issue, I installed Arch and now my Manjaro refuses to boot. And as Arch is now my main OS, I can't easily reproduce the issue. I'm sorry for this "false alarm". I didn't mean to work you uselessly. Hmm. I just wished to help firejail become even better.

As for my part, Manjaro is based on Arch, and yes, it gave the exact same error message on the first run. But after I ran it as you said, netblue30, without firejail, and then VirtualBox opened. After this and changing to the linux-lts kernel (I don't know how much this can affect VirtualBox and firejail) VirtualBox opened in firejail like `firejail virtualbox %u`. So it seems that either the linux-lts kernel fixed that, or I had to run VirtualBox once outside firejail. One of these fixed my issue.

Thank you really much again.

@Fred-Barclay : great to see you here too! :) Believe it or not, I learned Python3 in an afternoon/evening (I built upon my programming knowledge that I was thought instead of just using certain programs in high school), and started coding my homework right away. Very good indeed. :)
